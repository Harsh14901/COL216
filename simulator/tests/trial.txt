addi $s0, $zero, 1000
addi $t0, $zero, 50
sw $t0, 0($s0)
lw $t1, 0($s0)
sw $t0, 0($s0)
lw $t2, 0($s0)
sw $t0, 4($s0)
lw $t3, 0($s0)
sw $t0, 0($s0)

lw $t1, 4($s0)
lw $t2, 4($s0)
lw $t3, 4($s0)
