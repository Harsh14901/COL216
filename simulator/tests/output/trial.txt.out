[+] Compiling file : tests/input/trial.txt
[+] Program compiled successfully !
[+] Executing program ...
[+] Executing in NON-BLOCKING MODE
---------- Execution logs ----------
[$] CYCLE 1-1 :
[#] Executing current instruction -> addi $s0, $zero, 1000
[#] Register contents -
	 $zero :          0		   $s0 :       1000
	   $at :          0		   $s1 :          0
	   $v0 :          0		   $s2 :          0
	   $v1 :          0		   $s3 :          0
	   $a0 :          0		   $s4 :          0
	   $a1 :          0		   $s5 :          0
	   $a2 :          0		   $s6 :          0
	   $a3 :          0		   $s7 :          0
	   $t0 :          0		   $t8 :          0
	   $t1 :          0		   $t9 :          0
	   $t2 :          0		   $k0 :          0
	   $t3 :          0		   $k1 :          0
	   $t4 :          0		   $gp :          0
	   $t5 :          0		   $sp :    1048576
	   $t6 :          0		   $fp :          0
	   $t7 :          0		   $ra :          0
[#] Row buffer updates:
[#] Memory updates:
[#] Remarks: 
============================================================
[$] CYCLE 2-2 :
[#] Executing current instruction -> lw $a0, 0($zero)
[#] Register contents -
[#] Row buffer updates:
[#] Memory updates:
[#] Remarks: 
	DRAM request issued
============================================================
[$] CYCLE 3-12 :
[#] Executing current instruction -> 
[#] Register contents -
[#] Row buffer updates:
[#] Memory updates:
[#] Remarks: 
	DRAM: activating row 0
============================================================
[$] CYCLE 13-14 :
[#] Executing current instruction -> 
[#] Register contents -
	 $zero :          0		   $s0 :       1000
	   $at :          0		   $s1 :          0
	   $v0 :          0		   $s2 :          0
	   $v1 :          0		   $s3 :          0
	   $a0 :          0		   $s4 :          0
	   $a1 :          0		   $s5 :          0
	   $a2 :          0		   $s6 :          0
	   $a3 :          0		   $s7 :          0
	   $t0 :          0		   $t8 :          0
	   $t1 :          0		   $t9 :          0
	   $t2 :          0		   $k0 :          0
	   $t3 :          0		   $k1 :          0
	   $t4 :          0		   $gp :          0
	   $t5 :          0		   $sp :    1048576
	   $t6 :          0		   $fp :          0
	   $t7 :          0		   $ra :          0
[#] Row buffer updates:
[#] Memory updates:
[#] Remarks: 
	DRAM: Read 0 from row buffer @ 0 to register: $a0
============================================================
---------- Execution statistics ----------
[#] Clock cycles: 14
[#] Number of row buffer updates: 1
[#] Updated memory : 
[+] Program terminated
